docker/Dockerfile.webapp

```diff
@@ -1,15 +1,20 @@
 FROM local/socorro_base
 
-# Install nodejs and npm
-RUN DEBIAN_FRONTEND=noninteractive apt-get install -y npm nodejs nodejs-legacy
+# Install nodejs and npm from Nodesource's 8.x branch
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y curl
+RUN DEBIAN_FRONTEND=noninteractive curl -sL https://deb.nodesource.com/setup_8.x | bash -
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y nodejs
 
-ENV LESS_BINARY /usr/local/lib/node_modules/socorro-webapp-django/node_modules/.bin/lessc
-ENV UGLIFYJS_BINARY /usr/local/lib/node_modules/socorro-webapp-django/node_modules/.bin/uglifyjs
-ENV CSSMIN_BINARY /usr/local/lib/node_modules/socorro-webapp-django/node_modules/.bin/cssmin
+COPY ./webapp-django/package.json /webapp-frontend-deps/package.json
+COPY ./webapp-django/package-lock.json /webapp-frontend-deps/package-lock.json
+RUN cd /webapp-frontend-deps/ && npm install
 
 COPY . /app/
 
-RUN cd /app/webapp-django/ && npm install -g
+ENV LESS_BINARY /webapp-frontend-deps/node_modules/.bin/lessc
+ENV UGLIFYJS_BINARY /webapp-frontend-deps/node_modules/.bin/uglifyjs
+ENV CSSMIN_BINARY /webapp-frontend-deps/node_modules/.bin/cssmin
+ENV NPM_ROOT_PATH /webapp-frontend-deps/
```

Comment:
I moved these mostly for better cache usage, since we don't really need to redownload the frontend or app directory files if they change.